modules/documents/client/directives/documents.confirm.unpublish.directive.js

```diff
@@ -0,0 +1,302 @@
+                promises.push(new Promise (function (resolve, reject) {
+                  checkFolderForContent(item)
+                    .then(function() {
+                      item.canBeUnpublished = (!item.hasPublished && item.userCanUnpublish);
+                      if (item.canBeUnpublished) {
+                        self.unpublishableFolders.push(item);
+                      } else {
+                        setReasonForItem(item);
+                      }
+                      return resolve(item);
+                    }).catch(function (err) {
+                      self.errMsg = err.message;
+                      return reject(err);
+                    });
+                }));
+              });
+              // Parse files and determine eligibility for unpublishing.
+              _.forEach(self.files, function(item) {
+                promises.push(new Promise (function (resolve, reject) {
+                  checkFileForCollections(item, self.files)
+                    .then(function() {
+                      item.canBeUnpublished = (item.restrictingCollections.length < 1 && item.userCanUnpublish);
+                      if (item.canBeUnpublished) {
+                        self.unpublishableFiles.push(item);
+                      } else {
+                        setReasonForItem(item);
+                      }
+                      return resolve(item);
+                    }).catch(function (err) {
+                      self.errMsg = err.message;
+                      return reject(err);
+                    });
+                }));
+              });
+              return Promise.all(promises);
+            }
+
+            // Collect unpublishable folders and files.
+            function collect(items) {
+              // Filter out items that are already unpublished
+              if (!items ) {
+                return [];
+              }
+              // Service may be invoked on a single folder or file ....
+              items = Array.isArray(items) ? items : [ items ];
+              var results = _.map(items, function(item) {
+                var fClone = _.clone(item /* shallow clone */);
+                fClone.reasons = [];
+                // Clones has several new variables to simplify this directive.
+                if (fClone.type === 'File') {
+                  fClone.userCanUnpublish = fClone.userCan.unPublish;
+                  fClone.restrictingCollections = [];
+                  fClone.type = ['png','jpg','jpeg'].indexOf(fClone.internalExt) > -1 ? 'Picture' : 'File';
+                } else {
+                  fClone.userCanUnpublish = self.project.userCan.manageFolders;
+                  fClone.type = 'Folder';
+                  fClone.displayName = item.model.name;
+                  fClone.isPublished = item.model.published;
+                }
+                fClone.canBeUnpublished = fClone.userCanUnpublish && !fClone.isPublished;
```

Comment:
This seems to indicate a file/folder can be unpublished if:
* the current user has permission to unpublish - it makes sense
* the file/folder is not published - *this seems odd*
What am I missing here?

Reply:
Not missing anything; just a redundancy I forgot to remove when refactoring this code block. I'll snip it and re-commit.